campaignion_tracking/campaignion_tracking.module

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * @file
+ * Hook implementations for the Campaignion Tracking module.
+ */
+
+use Drupal\campaignion_action\Loader;
+use Drupal\little_helpers\Webform\Submission;
+use Drupal\little_helpers\Webform\Webform;
+
+/**
+ * Implements hook_preprocess_html().
+ */
+function campaignion_tracking_preprocess_html(&$vars) {
+  $module_path = drupal_get_path('module', 'campaignion_tracking');
```

Comment:
In general I prefer `#attached` over `drupal_add_*`. This would require moving this into `hook_page_build()` or something similar.

Reply:
Is there a reason to prefer `#attached`?

Reply:
Yes. It makes your function a pure function instead of having side-effects, which means you might be able to actually test it.

Reply:
If that is the goal, then you cannot use `$element['#attached']['js'][]` but need define a key, I guess. Otherwise it's not pure...

Reply:
You should be able to translate this directly into:

```php
$element['#attached'][$path] = $options;
```

See [drupal_process_attached()](https://devdocs.io/drupal~7/includes-common.inc/function/drupal_process_attached/7.x)